doc/build_apps/js_app_bundle.rst

```diff
@@ -130,6 +130,21 @@ In many places where timestamps are desired, they should come from the outside w
 
 To ease porting of existing apps, and for logging scenarios, there is an option to retrieve the current time from the host. When the executing CCF node is run by an honest operator this will be kept up-to-date, but the accuracy of this is not covered by any attestation and as such these times should not be relied upon. To enable use of this untrusted time, call ``ccf.enableUntrustedDateTime(true)`` at any point in your application code, including at the global scope. After this is enabled, calls to ``Date.now()`` will retrieve the current time as specified by the untrusted host. This behaviour can also be revoked by a call to ``ccf.enableUntrustedDateTime(false)``, allowing the untrusted behaviour to be tightly scoped, and explicitly opted in to at each call point.
 
+Execution metrics
+~~~~~~~~~~~~~~~~~
+
+By default the CCF JS runtime will print a log line for each completed JS request. This lists the request path and response status as well as how long the request took to execute. Each line also includes a ``[js]`` tag so that they can easily be filtered and sent to an automated monitoring system. These lines have the following format:
+
+.. code-block::
+
+    <timestamp> [info ][js] <...> | JS execution complete: Method=GET, Path=/app/make_randoms, Status=200, ExecMilliseconds=30
```

Comment:
Because this logging and its specific format are intended to be consumed by other tools, I suggest we capture it in at least one end to end test, so as to avoid any intentional regressions or changes in the future.

Reply:
I've added an end-to-end for this. We used to have a `get_logs()` that was called in every `stop()`, and actually returned the filtered errors. I've refactored this so that `get_logs()` returns the path to the out and err files (copied from a remote if necessary), so they can be used by this test, and the places that were actually using the error logs (`network.py`) now read and filter themselves. This almost certainly has unintended consequences for how we detect node shutdown errors.